neutron/plugins/ml2/drivers/openvswitch/agent/ovs_neutron_agent.py

```diff
@@ -1654,10 +1654,23 @@ def process_network_ports(self, port_info, ovs_restarted):
         self._add_port_tag_info(need_binding_devices)
         if security_disabled_ports:
             added_ports -= set(security_disabled_ports)
-        self.sg_agent.setup_port_filters(added_ports,
-                                         port_info.get('updated', set()))
+
+        # The following call was split into two calls so that added_ports
+        # can be bound quicker.
+        #self.sg_agent.setup_port_filters(added_ports,
+        #                                 port_info.get('updated', set()))
+
+        devices_to_refilter, global_refresh_firewall = (
+            self.sg_agent.setup_port_filters_part_1(added_ports))
+
         failed_devices['added'] |= self._bind_devices(need_binding_devices)
```

Comment:
Nate, if you have time and you could verify that this change does not open a security hole ... that would be great.   Since both yourself and Richard have questioned whether this change opens a security hole, the "safe" thing would be to just abandon this change.  If you don't have time to test this, that is fine... I'll just abandon this patch.